src/execution_plan/ops/op_node_by_label_scan.c

```diff
@@ -126,13 +132,10 @@ static inline void _ResetIterator(NodeByLabelScan *op) {
 		NodeID maxId = op->id_range->include_max ? op->id_range->max : op->id_range->max - 1 ;
 		RG_MatrixTupleIter_iterate_range(&op->iter, minId, maxId);
 	} else {
-		// id_range is NULL, this operation must have been freed previously
-		// rebuild the range iterator
-		GraphContext *gc = QueryCtx_GetGraphCtx();
-		Schema *schema = GraphContext_GetSchema(gc, op->n.label, SCHEMA_NODE);
-		if(!schema) return; // invalid schema, our consume function is NOP
+		_update_label_id(op);
```

Comment:
I feel like calling `_update_label_id` at runtime is redundant and should be removed, please validate this assumption.